staging/src/k8s.io/client-go/tools/leaderelection/resourcelock/configmaplock.go

```diff
@@ -87,7 +90,15 @@ func (cml *ConfigMapLock) Update(ler LeaderElectionRecord) error {
 		return err
 	}
 	cml.cm.Annotations[LeaderElectionRecordAnnotationKey] = string(recordBytes)
-	cml.cm, err = cml.Client.ConfigMaps(cml.ConfigMapMeta.Namespace).Update(cml.cm)
+
```

Comment:
Why are we patching vs. doing the full overwrite?  For things the an annotation I don't see the need todo this.

Reply:
It was related to Jason's comment on https://github.com/kubernetes/kubeadm/issues/1097: `Creating a new configmap lock where the annotation does not already exist causes the configmap Data to be clobbered.`. I could give it a go and see how to improve it as well, but https://github.com/kubernetes/kubernetes/pull/76821 looks to me like a better alternative to leader election.